src/backend/joanie/payment/models.py

```diff
@@ -1 +1,430 @@
+    )
+
+    translations = TranslatedFields(
+        context=models.JSONField(
+            _("context"),
+            help_text=_("Context used to generate a credit note PDF document"),
+            null=False,
+            blank=False,
+            editable=False,
+        )
+    )
+
+    class Meta:
+        db_table = "joanie_transaction"
+        verbose_name = _("Transaction")
+        verbose_name_plural = _("Transactions")
+
+    def __str__(self):
+        transaction_type = "Credit" if self.is_credit else "Debit"
+        return f"{transaction_type} transaction {self.reference}"
+
+    def _get_context_for_locale(self, language_code=None):
+        """
+        Generate credit note context
+        """
+        language_code = language_code or get_language()
+        with switch_language(self.invoice, language_code):
+            vat = D(settings.JOANIE_VAT)
+            total_amount = -1 * self.amount
+            vat_amount = total_amount * vat / 100
+            net_amount = total_amount + vat_amount
+            currency = settings.JOANIE_CURRENCY[1]
+
+            return {
+                "metadata": {
+                    "reference": self.reference,
+                    "issued_on": str(self.created_on),
+                    "type": "credit_note",
+                },
+                "order": {
+                    # pylint: disable=no-member
+                    **self.invoice.context["order"],
+                    "amount": {
+                        "currency": currency,
+                        "subtotal": str(net_amount),
+                        "total": str(total_amount),
+                        "vat_amount": str(vat_amount),
+                        "vat": str(vat),
+                    },
+                },
+            }
+
+    def _create_localized_contexts_for_credit(self):
+        """
+        Generate a PDF document for a credit transaction.
+
+        This method should be called only when a credit transaction is created.
+        """
+        if self.is_credit is False:
+            raise ValueError("Transaction instance is not a credit.")
+
+        for language, __ in settings.LANGUAGES:
+            localized_context = self._get_context_for_locale(language)
+            # pylint: disable=no-member
+            self.translations.create(language_code=language, context=localized_context)
+
+    def get_credit_note(self):
+        """
+        Generate a document related to the credit note instance.
+        """
+        if self.is_credit is False:
+            raise ValueError("Transaction instance is not a credit.")
+
+        # pylint: disable=no-member
+        document = InvoiceDocument(context_query=self.context)
+        return document.create(persist=False)
+
+    @property
+    def document(self):
+        """
+        Return the credit note document if the transaction is a credit
+        otherwise return the related invoice document.
+        """
+        if self.is_credit:
```

Comment:
let's discuss this irl

Reply:
From now, transaction model is only in charge of register financial transaction. Those transactions can be reconcilied to an invoice.